xinference/model/embedding/core.py

```diff
@@ -224,6 +226,28 @@ def to(self, *args, **kwargs):
             )
 
     def create_embedding(self, sentences: Union[str, List[str]], **kwargs):
+        # Check if sentences is a two-dimensional list of integers
+        if isinstance(sentences, list) and all(
```

Comment:
I recommend to put all these codes into a function to explicitly explain what it is dong.

```
def _fix_langchain_openai_inputs(self, sentences: xxx):
    import tiktoken  # put import here, tiktoken is not a must-have dependency

    # check xxx
    if isinstance xxx
    
```

Reply:
ok

Reply:
The import needs to be moved like I said, or all CI would fail.

Reply:
OK